lib/level_db.rb

```diff
@@ -113,6 +113,24 @@ def each(options={}, &block)
     def snapshot
       Snapshot.new(@db)
     end
+
+    def compact_range(options={})
+      from = options[:from]
+      to = options[:to]
+      @db.compact_range(encode_key(from), encode_key(to))
+    end
+
+    def full_compaction
+      sstables = @db.get_property(SSTABLES_PROPERTY)
+      pre_compaction_sstables = nil
+      until pre_compaction_sstables == sstables
+        pre_compaction_sstables = sstables
+        compact_range
+        sstables = @db.get_property(SSTABLES_PROPERTY)
+      end
```

Comment:
But then I should be able to compare the number of files per level like
```ruby
def num_files_per_level
  7.times.map { |i| @db.get_property("leveldb.num-files-at-level#{i}") }
end

pre_compaction = num_files_per_level
compact_range
post_compaction = num_files_per_level
converged = post_compaction == pre_compaction
```
That should be "safe", right? We'll obviously keep going if we're writing to the DB while compacting, but that should be true regardless of which method we use. But using this method, at least we won't be exposed to

> the same number of SSTables were added that were compacted away between two calls

if I'm not mistaken?